[9.0][MIG] l10n_es_aeat_mod111: Migration to 9.0

pedrobaeza opened the pull request:
* Added tests

cc @Tecnativa

antespi committed:
[MIG] l10n_es_aeat_mod111: Migration to 9.0

* Added tests

pedrobaeza commented:
Revisa por favor @cubells

cubells reviewed (approved):
Revisado el código y testeado en el runbot

pedrobaeza committed:
[FIX] l10n_es_aeat_mod111: Add multi-company security rule

rafaelbn reviewed (approved):
Tested

pedrobaeza committed:
[FIX] l10n_es_aeat_mod111: Avoid error on multi-company envs

With the limitation of ORM with tax_line_ids in @api.depends,
we avoid the access error this way

pedrobaeza committed:
[FIX] l10n_es_aeat_mod111: Remove incorrect code

pedrobaeza commented:
Paso a mergear después de un buen tiempo de explotación.
